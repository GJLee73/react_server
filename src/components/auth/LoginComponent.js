import React from 'react';
import {Link} from 'react-router-dom';

class LoginComponent extends React.Component {
	constructor(props){
		super(props);
		this.state = {
			userId:"",
			password:""
		};
		this.handleChange = this.handleChange.bind(this);
		this.handleLogin = this.handleLogin.bind(this);
		this.handleKeyPress = this.handleKeyPress.bind(this);
	}

	/** 
	 * 로그인 후 성공이 아니면 패스워드를 지워준다.
	 */
	handleLogin(){
		let userId = this.state.userId;
		let pw = this.state.password;
		console.log(userId)
		this.props.onLogin(userId, pw).then(
			(success) => {
				if(!success){
					this.setState({
						password: ''
					})
				}
			}
		).catch(err=>{
			console.log(err);
		})
	}

	handleChange(e){ 
		let nextState = {};
		nextState[e.target.name] = e.target.value;
		this.setState(nextState);
	}

	handleKeyPress(e){
		if(e.charCode===13){
			
		    this.handleLogin();
		
        }
    }

	render(){
	 
    
			

		return (
			<div className="container auth">
				<Link className="logo" to="/">5437</Link>
				<div className="card">
					<div className="header blue white-text center">
						<div className="card-content">Login</div>
					</div>
					<div className="card-content">
						<div className="row">
							
						<div className="input-field col s12 username">
							<label>Id</label>
							<input
							name="userId"
							type="text"
							className="validate"
							onChange={this.handleChange}
							value={this.state.userId}/>
						</div>
						<div className="input-field col s12">
							<label>Password</label>
							<input
							name="password"
							type="password"
							className="validate"
							onChange={this.handleChange}
							value={this.state.password}
							onKeyPress={this.handleKeyPress}/>
						</div>
					
							<a className="waves-effect waves-light btn" onClick={this.handleLogin}>SUBMIT</a>
						</div>
					</div>

					<div className="footer">
						<div className="card-content">
							<div className="right">
								<a href="/register">Create an account</a>
								{/* materializecss가 selecter를 사용하려면 script를 html에서 사용하는데 이걸 하기 위해서는
								refresh시켜주어야만 함. */}
								/	
								<Link to='/find'>아이디 비밀번호찾기</Link>
							</div>
						</div>
					</div>
				</div>
			</div>
		);
	}
}

export default LoginComponent;