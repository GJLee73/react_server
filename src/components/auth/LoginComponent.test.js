import React from 'react';
import { shallow} from 'enzyme';
import LoginComponent from './LoginComponent'
import {MemoryRouter} from 'react-router-dom'

describe('LoginComponent test',()=>{
    var handleChange = jest.spyOn(LoginComponent.prototype,'handleChange')
    var handleKeyPress = jest.spyOn(LoginComponent.prototype,'handleKeyPress')
    var handleLogin = jest.spyOn(LoginComponent.prototype,'handleLogin')
    var onLogin = jest.fn(()=>{
        if(wrapper.state()=={userId:'g1',password:'54375437!'})
        return Promise.resolve()
        else return Promise.reject('login fail')
    })
    var wrapper = shallow(<LoginComponent onLogin={onLogin}/>)

    var successId ={target:{value:'g1',name:'userId'}}
    var failId = {target:{value:'g2',name:'userId'}}
    var successPwd ={target:{value:'54375437!',name:'password'}}
	var failPwd ={target:{value:'....',name:'password'}}
    var inputs = wrapper.find('input')
    describe('function test',()=>{
      
        it('handleChange should work',()=>{
		
            inputs.at(0).simulate('change',successId)
            inputs.at(1).simulate('change',successPwd)
            expect(handleChange.mock.calls.length).toBe(2)
            expect(wrapper.state()).toEqual({"password": "54375437!", "userId": "g1"})
       })
       it('will login if it had true id,pwd',()=>{
           inputs.at(1).simulate('keypress',{charCode:13})
           wrapper.find('a').at(0).simulate('click')
           expect(onLogin.mock.calls.length).toBe(2)
       })
       it('can not login when it submit wrong id,pwd',()=>{
        inputs.at(0).simulate('change',failId)
        inputs.at(1).simulate('change',failPwd)

        inputs.at(1).simulate('keypress',{charCode:13})
        wrapper.find('a').at(0).simulate('click')
        expect(onLogin.mock.calls.length).toBe(4)
    })

    })
    it('snapshot test',()=>{
      expect(wrapper).toMatchSnapshot()
    })
    it('for 100% branch',()=>{
        inputs.at(1).simulate('keypress',{charCode:1})
    })
    
})