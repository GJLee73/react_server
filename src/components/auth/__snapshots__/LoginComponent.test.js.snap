// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent test snapshot test 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropTransition={
    Object {
      "mountOnEnter": true,
      "timeout": 150,
    }
  }
  centered={false}
  fade={true}
  isOpen={false}
  keyboard={true}
  modalTransition={
    Object {
      "timeout": 300,
    }
  }
  onClosed={[Function]}
  onOpened={[Function]}
  role="dialog"
  style={
    Object {
      "marginTop": "100px",
      "width": "350px",
    }
  }
  zIndex={1050}
>
  <div
    className="d-flex flex-row"
    style={
      Object {
        "paddingLeft": "50px",
        "paddingTop": "40px",
      }
    }
  >
    <h2>
      Splendy
    </h2>
    <h3
      style={
        Object {
          "margin": 0,
          "paddingLeft": "10px",
          "paddingTop": "5px",
        }
      }
    >
      로그인
    </h3>
  </div>
  <ModalBody
    style={
      Object {
        "paddingLeft": "30px",
        "paddingTop": "0",
      }
    }
    tag="div"
  >
    <div
      className="card-body"
      style={
        Object {
          "paddingBottom": "0",
        }
      }
    >
      <h5
        className="card-title"
      >
        Id
      </h5>
      <input
        className="validate"
        name="userId"
        onChange={[Function]}
        type="text"
        value="g2"
      />
    </div>
    <div
      className="card-body"
      style={
        Object {
          "paddingBottom": "0",
        }
      }
    >
      <h5
        className="card-title"
      >
        Password
      </h5>
      <input
        className="validate"
        name="password"
        onChange={[Function]}
        onKeyPress={[Function]}
        type="password"
        value="...."
      />
    </div>
    <div
      className="card-body"
    >
      <button
        className="btn btn-danger"
      >
        <a
          className="waves-effect waves-light btn"
          onClick={[Function]}
        >
          SUBMIT
        </a>
      </button>
    </div>
  </ModalBody>
  <ModalFooter
    tag="div"
  >
    <div
      className="card-content"
    >
      <div
        className="right d-flex "
      >
        <div
          className="text-secondary"
          onClick={[Function]}
        >
          Create an account
        </div>
        /
        <div
          className="text-secondary"
          onClick={[Function]}
        >
          아이디 비밀번호찾기
        </div>
      </div>
    </div>
  </ModalFooter>
</Modal>
`;
